youtube_dl/extractor/vidlii.py

```diff
@@ -0,0 +1,129 @@
+    float_or_none, urljoin)
+
+
+class VidliiIE(InfoExtractor):
+    _VALID_URL = r'(?:https*?:\/\/)*(?:www\.)*vidlii.com\/watch\?v=(?P<id>[^?\s]{11})'
+    _TESTS = [{
+        'url': 'https://www.vidlii.com/watch?v=tJluaH4BJ3v',
+        'md5': '9bf7d1e005dfa909b6efb0a1ff5175e2',
+        'info_dict': {
+            'id': 'tJluaH4BJ3v',
+            'title': 'Vidlii is against me',
+            'description': 'I have HAD it. Vidlii does not like me. I have tried to uplaod videos and submit them to the '
+                           'contest and no ne of my videos show up so maybe it is broken for everyone else but this one was '
+                           'trying to submit it because I wanted to submit to the contest :) Tanks I hope the website is '
+                           'fixed PS: Jan you are cool please add my video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/tJluaH4BJ3v.jpg',
+            'uploader': 'APPle5auc31995',
+            'url': 'https://cdn.vidlii.com/videos/tJluaH4BJ3v.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/APPle5auc31995',
+            'upload_date': '20171107',
+            'categories': 'News & Politics',
+            'tags': ['Vidlii', 'Jan', 'Videogames'],
+            'duration': 212,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'https://www.vidlii.com/watch?v=vBo2IcrwOkO',
+        'md5': 'b42640a596b4dc986702567d49268963',
+        'info_dict': {
+            'id': 'vBo2IcrwOkO',
+            'title': '(OLD VIDEO) i like youtube!!',
+            'description': 'Original upload date:<br />\nMarch 10th 2011<br />\nCredit goes to people who own content in the video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/vBo2IcrwOkO.jpg',
+            'uploader': 'MyEditedVideoSpartan',
+            'url': 'https://cdn.vidlii.com/videos/vBo2IcrwOkO.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/MyEditedVideoSpartan',
+            'upload_date': '20171011',
+            'categories': 'Film & Animation',
+            'tags': None,
+            'duration': 34,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # extract basic properties of video
+        title = self._html_search_regex(r'<h1>(.+?)</h1>', webpage, 'title', default=None) or self._html_search_regex(
+            r'<title>([^<]+?)</title>', webpage, 'title', default="").replace(
+            " - VidLii", "") or self._html_search_meta('twitter:title', webpage, 'title',
+                                                       default="").replace(" - VidLii", "")
```

Comment:
No. `title` is mandatory. Code duplication. Move all `_html_search_regex` into a single call.

Reply:
Please tell us exactly how we should handle it:

a) We combine the regular expressions into one single call, then we cannot make the replacement anymore such that tests would fail if any fallback is used. Because fallbacks contain the title plus the suffix " - VidLii". That's the reason why we remove the suffix.

b) We change the order such that our first extractor (without replace) now is used as the last one. This means, if the first two extractors cannot find any title, ```None``` is returned, indicating that a mandatory field is missing. In this case the ```self._html_search_regex``` calls cannot be merged into a single one. See following code:

```
        title = (self._html_search_regex(r'<title>([^<]+?)</title>', webpage, 'title', default="") 
or self._html_search_meta('twitter:title', webpage, 'title', default="")).replace(" - VidLii","") 
or self._html_search_regex(r'<h1>(.+?)</h1>', webpage, 'title', default=None)
```

c) ..do you have any better suggestion?

Reply:
We now implemented solution b). If you do not like it, please tell us better suited alternatives. Thank you for your helpful feedback!